atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -165,6 +166,34 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
     public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition) {
         StartIdentifiedAtlasDbTransactionResponse transactionResponse
                 = timelockService.startIdentifiedAtlasDbTransaction();
+        return wrap(condition, transactionResponse);
+    }
+
+    @Override
+    public List<TransactionAndImmutableTsLock> setupRunTaskWithConditionThrowOnConflictBatch(
+            List<PreCommitCondition> conditions) {
+        // now, this can throw if any fail, but the resources are cleaned up (no promises
+        // about the passed in conditions though
```

Comment:
I think we should make this do promises about conditions though, it feels like much better API.